compiler/test/dotty/tools/vulpix/FileDiff.scala

```diff
@@ -34,15 +35,20 @@ object FileDiff {
     outFile.writeAll(content.mkString("", EOL, EOL))
   }
 
-  def checkAndDump(sourceTitle: String, actualLines: Seq[String], checkFilePath: String): Boolean =
+  def checkAndDump(sourceTitle: String, actualLines: Seq[String], checkFilePath: String): Boolean = {
+    val outFilePath = checkFilePath + ".out"
     FileDiff.check(sourceTitle, actualLines, checkFilePath) match {
       case Some(msg) =>
-        val outFilePath = checkFilePath + ".out"
         FileDiff.dump(outFilePath, actualLines)
         println(msg)
         println(FileDiff.diffMessage(checkFilePath, outFilePath))
         false
       case _ =>
+        val jOutFilePath = Paths.get(outFilePath)
+        if (Files.exists(jOutFilePath))
+          try { Files.delete(jOutFilePath) } catch { case _: Exception => () }
```

Comment:
What will happen if we leave out the code here? If they are really needed, it seems the best place is inside the method `dump`? BTW, `Files. deleteIfExists` will make the code shorter (maybe remove the try/catch).

Reply:
Here is a scenario to explain why I would not leave out the highlighted code.

Given the test case from command `sbt "testOnly dotty.tools.dotc.PrintingTest"`,
1) I run the above command and it fails.
    I can compare the output file (suffx `.check.out`) with the check file (suffix `.check`).
2) I fix the issue (and compile the modified source files).
4) I rerun the above command and it succeeds.
    The output file (suffx `.check.out`) is **still** there (generated by step 1) despite a successful execution.

Yes, using `Files.deleteIExists` is better.